Source/Core/Core/HW/WiimoteEmu/EmuSubroutines.cpp

```diff
@@ -240,13 +241,26 @@ void Wiimote::HandleRequestStatus(const OutputReportRequestStatus&)
   // Max battery level seems to be 0xc8 (decimal 200)
   constexpr u8 MAX_BATTERY_LEVEL = 0xc8;
 
-  m_status.battery = u8(std::lround(m_battery_setting.GetValue() / 100 * MAX_BATTERY_LEVEL));
-
+#ifdef SYSTEM_BATTERY_LEVEL_AVAILABLE
+  std::optional<float> battery;
+#endif
   if (Core::WantsDeterminism())
   {
     // One less thing to break determinism:
     m_status.battery = MAX_BATTERY_LEVEL;
   }
+#ifdef SYSTEM_BATTERY_LEVEL_AVAILABLE
+  else if (m_system_battery_setting.GetValue() && (battery = GetSystemBatteryLevel()))
+  {
+    // HOME Menu treats this as a percentage, if weirdly rounded.
+    // 100 shows as 4/4, 75 as 3/4, 51 (not 50) as 2/4.
+    m_status.battery = u8(std::lround(*battery * 100));
+  }
+#endif
+  else
+  {
+    m_status.battery = u8(std::lround(m_battery_setting.GetValue() / 100 * MAX_BATTERY_LEVEL));
```

Comment:
Note: I have left this code as it is, but I think it is wrong. MAX_BATTERY_LEVEL is way beyond 100% if we use HOME menu as a reference, so even if you set the m_battery_setting to 50% currently, it still shows up as 4/4 bars.

Reply:
The Remote hardware definitely produces values around 0xc8 (decimal 200) for new batteries.

Perhaps the HOME menu has been designed to show 4/4 bars for partially drained batteries to comfort players. 4/4 is more of a "you're all good" rather than a "it rounds up to 100%".

Having said that, I don't know the best way to implement this feature when the visual does not match the underlying value nicely.

Reply:
I think I will check some other Wii software that displays the battery, to see if they use the value differently.

Reply:
It may be designed to show a visual change only when the level comes below the standardized value of the required 1.5V AA batteries, which usually come at 1.65V when new/fully charged and that would produce a total of 3.3V which seems to be the optimal value considering the Wii Remote IR camera is labeled as _3.3V_

For example it could be something like: 

- 200 = 3.3 VDC
- 100 = 3.0 VDC
- 0 = 2.7 VDC

Reply:
I tried Metroid Prime 3 and it seems to display the battery value bars exactly the same as the HOME menu, at least for 100, 75, 51 and 50.

You might be onto something with the voltage thing. Another option would be for the settings to have 200% as the maximum and default value, corresponding to the integer 200.

Reply:
I could adjust the existing setting in this PR, or make another PR, if you think it's a good idea.

Reply:
I was wrong, it's not really percentage-based.

Reply:
I don't like the "hacky math".

I think it would make more sense to map the system's 0-100% level directly to the 0-200 value produced by a Wii Remote.

Why must we hack the value attempting to produce linear output on the "4 bars" when that is not even how the real system works?

Reply:
Well, yes, you could do that, but it would also make the feature completely useless.

Reply:
> Why must we hack the value attempting to produce linear output on the "4 bars" when that is not even how the real system works?

Battery voltage doesn't decline linearly with battery capacity. I should probably be trying to emulate that behaviour rather than doing this fudge though.

Reply:
I'd be happier with emulating a battery voltage decline.

Reply:
I looked up an example AA battery voltage curve and made a function that roughly fit it, but it is no better than the identity function in terms of making a given percentage produce a sensible UI result. Now I am wondering what the Wiimote byte actually means, and how the HOME Menu interprets it.